src/main/groovy/org/jetbrains/intellij/tasks/RunPluginVerifierTask.groovy

```diff
@@ -0,0 +1,354 @@
+    }
+
+    @Input
+    @Optional
+    String getVerificationReportsDir() {
+        return verificationReportsDir
+    }
+
+    void setVerificationReportsDir(String verificationReportsDir) {
+        this.verificationReportsDir = verificationReportsDir
+    }
+
+    @Input
+    @Optional
+    String getJbrVersion() {
+        return jbrVersion
+    }
+
+    void setJbrVersion(String jbrVersion) {
+        this.jbrVersion = jbrVersion
+    }
+
+    @Input
+    @Optional
+    String getRuntimeDir() {
+        return runtimeDir
+    }
+
+    void setRuntimeDir(String runtimeDir) {
+        this.runtimeDir = runtimeDir
+    }
+
+    @Input
+    @Optional
+    String getExternalPrefixes() {
+        return externalPrefixes
+    }
+
+    void setExternalPrefixes(String externalPrefixes) {
+        this.externalPrefixes = externalPrefixes
+    }
+
+    @Input
+    @Optional
+    String getPluginsToCheckAllBuilds() {
+        return pluginsToCheckAllBuilds
+    }
+
+    void setPluginsToCheckAllBuilds(String pluginsToCheckAllBuilds) {
+        this.pluginsToCheckAllBuilds = pluginsToCheckAllBuilds
+    }
+
+    @Input
+    @Optional
+    String getPluginsToCheckLastBuilds() {
+        return pluginsToCheckLastBuilds
+    }
+
+    void setPluginsToCheckLastBuilds(String pluginsToCheckLastBuilds) {
+        this.pluginsToCheckLastBuilds = pluginsToCheckLastBuilds
+    }
+
+    @Input
+    @Optional
+    String getTeamCity() {
+        return teamCity
+    }
+
+    void setTeamCity(String teamCity) {
+        this.teamCity = teamCity
+    }
+
+    @Input
+    @Optional
+    String getTcGrouping() {
+        return tcGrouping
+    }
+
+    void setTcGrouping(String tcGrouping) {
+        this.tcGrouping = tcGrouping
+    }
+
+    @Input
+    @Optional
+    String getExcludedPluginsFile() {
+        return excludedPluginsFile
+    }
+
+    void setExcludedPluginsFile(String excludedPluginsFile) {
+        this.excludedPluginsFile = excludedPluginsFile
+    }
+
+    @Input
+    @Optional
+    String getDumpBrokenPluginList() {
+        return dumpBrokenPluginList
+    }
+
+    void setDumpBrokenPluginList(String dumpBrokenPluginList) {
+        this.dumpBrokenPluginList = dumpBrokenPluginList
+    }
+
+    @Input
+    @Optional
+    String getPluginsToCheckFile() {
+        return pluginsToCheckFile
+    }
+
+    void setPluginsToCheckFile(String pluginsToCheckFile) {
+        this.pluginsToCheckFile = pluginsToCheckFile
+    }
+
+    @Input
+    @Optional
+    String getSubsystemsToCheck() {
+        return subsystemsToCheck
+    }
+
+    void setSubsystemsToCheck(String subsystemsToCheck) {
+        this.subsystemsToCheck = subsystemsToCheck
+    }
+
+    @TaskAction
+    void runPluginVerifier() {
+        def extension = project.extensions.findByType(IntelliJPluginExtension)
+        def pluginFileName = "${extension.pluginName}-${project.version}"
+
+        if (!project.file("${project.buildDir}/distributions/${pluginFileName}.zip").exists()) {
```

Comment:
I don't think you should fail, I believe you should `SKIP`. For that you need accept plugin file and set it `SkipIfEmpty`